libraries/chain/controller.cpp

```diff
@@ -4472,6 +4470,11 @@ struct controller_impl {
                      shutdown();
                      break;
                   }
+                  // Break every ~500ms to allow other tasks (e.g. get_info, SHiP) opportunity to run. There is a post
+                  // for every incoming blocks; enough posted tasks to apply all blocks queued to the fork db.
+                  if (!replaying && fc::time_point::now() - start > fc::milliseconds(500))
+                     break;
```

Comment:
Should break or return? Not sure some of the code below still applies, like the ilog for `"successfully switched fork...`.

Reply:
Thanks. This should check `!switch_fork`. I think better to `break`. Currently nothing below will execute, but logically I think it makes more sense to `break`.